### ISSUE TITLE
Breakpoint set but not yet bound

### ISSUE THREAD

**Issue by dancomanlive** [Microsoft/vscode-react-native#625]
## Actual Behavior
1. Breakpoint does not stop code execution

## Expected Behavior
1. Breakpoint stops code execution

- Latest version of everything.
- "Breakpoint set but not yet bound" is displayed when hovering over the breakpoint.
- Debugger is working just fine, when pressing the pause button the execution stops.
- Console log shows that the code is running in the file where the breakpoint is set.

## Workaround
Just remove any of your breakpoints and add it again - breakpoints become red and stop code execution

**Comment by AylwSt01** [comment]
i'm getting this too... very annoying!

**Comment by novellizator** [comment]
So something definitelly with the newest update, since it worked in the past.

**Comment by sergey-akhalkov** [comment]
Hey guys, thanks for reaching us and sorry for the delayed response. We are going to investigate it asap and let you know if we have any questions or updates.

**Comment by novellizator** [comment]
Is there anything we can help with? I'll have to switch from VSCode unless this gets fixed :/

**Comment by sergey-akhalkov** [comment]
Hi @novellizator, yes, it would be great, because I've just tried to debug my iOS test app - it works as expected without any issues and I was able to debug my code. Going to verify on Android and then switch to Windows.

So could you guys please share with us you exact environment and provide a minimum working version of your code where the issue reproduces?

**Comment by edmundito** [comment]
I just setup VScode for the first time and ran into the same issue:

Vscode 1.20.1
RN Tools 0.5.7
RN 0.53.0

One unusual thing I noticed is that when I created the launch parameters I looked at `launch.json` and when I reviewed the docs for the Extension it recommended to add `internalDebuggerPort`, but I see this: 
![image](https://user-images.githubusercontent.com/168664/36386181-bf666838-1549-11e8-9a66-85efe0d76bd9.png)

Here is some additional information from the debug console:
```
OS: darwin x64
Adapter node: v7.9.0 x64
vscode-chrome-debug-core: 3.18.5
Starting debugger app worker.
Established a connection with the Proxy (Packager) to the React Native application
Debugger worker loaded runtime on port 4612
```

The "Debugger worker" port seems to be changing on every run.

**Comment by sergey-akhalkov** [comment]
Hi @edmundito, thanks for sharing the info!
> internalDebuggerPort

Hm, I have not touched it at all and it works just as it is.

>The "Debugger worker" port seems to be changing on every run.

The same is for me, so I don't think if it's a root cause, so I'm still trying to reproduce the issue.

**Comment by novellizator** [comment]
```Invalidating <RCTCxxBridge: 0x6040003c97e0> (parent: <RCTBridge: 0x6040000cae20>, executor: RCTWebSocketExecutor)
RCTLog.js:48
Debugger worker loaded runtime on port 33223
Module RCTImageLoader requires main queue setup since it overrides `init` but doesn't implement `requiresMainQueueSetup`. In a future release React Native will default to initializing all native modules on a background thread unless explicitly opted-out of.
RCTLog.js:48
Running application issue625 ({
RCTLog.js:48
    initialProps =     {
    };
    rootTag = 31;
})
Running application "issue625" with appParams: {"rootTag":31,"initialProps":{}}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF
```

Fun stuff: I added a console.warn to the click function so that it look like this

`  click() {
    console.warn("hahaha");
    this.setState({ clicks: ++this.state.clicks });
  }`

and set a breakpoint to this.setState(). The debugger didn't work.
But then, I set another breakpoint to the line with console.warn, and everything worked.

**Comment by dancomanlive** [comment]
![screen shot 2018-02-20 at 11 59 05 am 2](https://user-images.githubusercontent.com/16872821/36406943-8621bef0-1635-11e8-9e4f-fdfafde9ca2f.png)

Still not working. Tried on two different projects.

**Comment by novellizator** [comment]
can't it be some version of external dependency?

**Comment by sergey-akhalkov** [comment]
Hi @novellizator, could you please confirm - do I get it right that the issue has place since the new version of VSCode 1.20.0 or you have seen it before?

**Comment by novellizator** [comment]
I dont remember when the VSCode got updated, but it's a fairly recent issue. Can I somewhere find an older version of VSCode to try it out? (there isn't one on the homepage)

**Comment by sergey-akhalkov** [comment]
@novellizator, here is https://code.visualstudio.com/updates/v1_20 
For older version of VSCode you can found download links there, please let me know if you need anything else.

**Comment by novellizator** [comment]
are you sure there any any download links? don't see any... please send me
a link for macOS download of some older version....

**Comment by sergey-akhalkov** [comment]
>are you sure there any any download links?

yeah, here is 
<img width="906" alt="screen shot 2018-02-20 at 14 05 57" src="https://user-images.githubusercontent.com/4928157/36420827-48272cfe-1647-11e8-850a-a5701c952789.png">

Here is the link for Mac download (verions 1.19) https://vscode-update.azurewebsites.net/1.19.3/darwin/stable

**Comment by novellizator** [comment]
My bad, just quickly searched for "download" and didn't find anything relevant on the webpage.
Anyways, 1.19 works well. SO it only breaks on 1.20

**Comment by sergey-akhalkov** [comment]
@novellizator, got it, thanks, it's useful info. So could you please confirm if the issue reproduces on the demo app I've sent here https://github.com/Microsoft/vscode-react-native/issues/625#issuecomment-366719491? If so, could you please let me know what the line of the code you are setting a breakpoint to?

**Comment by sergey-akhalkov** [comment]
@novellizator, gotcha :)
Please let me know if the issue will happen for you again. I'm going to try one more thing but I'm still looking for reproducible demo / repro steps.

**Comment by sergey-akhalkov** [comment]
Hey guys, I've tried to reproduce the issue both for Android and iOS on macOS but with no luck. So I need your help to figure out what goes wrong. Could you please answer the following questions to make things clear:
1. Are you using TypeScript in your project?
2. Does the issue reproduces on the test app I've provided here https://github.com/Microsoft/vscode-react-native/issues/625#issuecomment-366719491 If not, please provide us a minimum working version of your app. @dancomanlive, could you please share the link to your app where the issue reproduces?
3. What OS are you using: macOS, Windows?
4. Does this issue have a stable repo? If so, could you please provide repro steps?
5. What launch configuration are you using, default or custom?
6. Have you ever seen this issue before VSCode version 1.20 or it reproduces for 1.20 only?
7. What version of `react-native-cli` are you using?
8. Please share exact version of your software you are using for development from [this issue template](https://github.com/Microsoft/vscode-react-native/issues/new)

**Comment by novellizator** [comment]
For what it's worth:
1. no
2. originally the issue reproduced on your code, also, now your code works
3. macOS
5. default
6. no, never before

**Comment by sergey-akhalkov** [comment]
@novellizator, thank you! Waiting for clarification from the others, just trying to find any leads that can help me to reproduce the issue.

**Comment by olegdizus** [comment]
@sergey-akhalkov 
1. No
2. Yes, issue was reproduced on your repo
3. macos High Sierra
4. Just cloned your repo, npm install, set breakpoints in App.js:27 and App.js:31 and faced this problem
5. Default
6. I think, I seen this issue before (september-october release maybe) but not sure.
7. react-native-cli: 2.0.1
8. 
- React-Native VS Code extension version: 0.6.0
- VSCode version: 1.20.1
- OS platform and version: macOS High Sierra 10.13.3
- NodeJS version: 9.5.0
- React Native version: 0.53.2 & 0.52.2 on my project
9. Reinstalled (settings not been erased) but got some problem.

**Comment by dancomanlive** [comment]
![screen shot 2018-02-21 at 11 26 49 am 2](https://user-images.githubusercontent.com/16872821/36461752-3fc5dcdc-16fa-11e8-9708-a7c8761d5992.png)
Version 1.21.0-insider (1.21.0-insider)
MacOS 10.13.3 (17D47)

**Comment by sergey-akhalkov** [comment]
Hey guys, thank you so much for sharing all the info. I've found 2 things (while debugging `vsrn625` I've provided previously) that really worrying me:
1. your `vscode-chrome-debug-core` version is `3.18.5`, but mine is `3.23.0`
2. your `rootTag` in appParams is `31`, but mine is `11`

So guys could you please totally remove React Native tools using this command `rm -rf ~/.vscode/extensions/vsmobile.vscode-react-native-0.6.1/` and then install it again to see if you have the same `vscode-chrome-debug-core` version and the same `rootTag` number?

Please let me know if you have any updates.

**Comment by YaroslavnaMatkova** [comment]
I've faced with the same issue. Debug console 
```
OS: win32 x64
Adapter node: v7.9.0 x64
vscode-chrome-debug-core: 3.23.0
Starting debugger app worker.
Established a connection with the Proxy (Packager) to the React Native application
Debugger worker loaded runtime on port 14765
Running application "ReactTestTask" with appParams: {"rootTag":1}. __DEV__ === true, development-level warning are ON, performance optimizations are OFF
```
 Re-installation of React Native tools helped.

**Comment by Sler69** [comment]
I recently installed the version 0.6.1 of vscode-react-native. The breakpoints are not working yet. 
react-native-tools: 0.6.1
vscode:1.20.1
OS: Ubuntu 17.10
react-native-cli: 2.0.1
node:v9.5.0

**Comment by f4z3k4s** [comment]
Same here. Breakpoints don't work. :(
react-native-tools: 0.6.1
vscode: 1.20.1
os: ubuntu 16.04
react-native-cli: 2.0.1
node: 8.9.4
react-native: 0.47

**Comment by sergey-akhalkov** [comment]
Hi @Sler69, @f4z3k4s, could you please let me know what is your version of `vscode-chrome-debug-core` (you can check it in debug console)? 
If this module has version `3.18.5` - please try to reinstall your extension as I've described [here](https://github.com/Microsoft/vscode-react-native/issues/625#issuecomment-367258000) and let me know if the issue still has place for you.

**Comment by f4z3k4s** [comment]
Hi @sergey-akhalkov, my `vscode-chrome-debug-core` version is : `3.23.0`.

**Comment by Sler69** [comment]
Hi @sergey-akhalkov, 
 vscode-chrome-debug-core version is : 3.23.0

**Comment by sergey-akhalkov** [comment]
Hey guys, I was able to reproduce it:
1. Completely remove RN Tools Extension *with all its dependencies* from VSCode
2. Install it again 
3. `Debug iOS`

The debugging actually works when I reset one of the debug breakpoints, but it's actually annoying. So I'm going to investigate the root cause of the issue and let you know if I have any updates.

**Comment by Gremlin80** [comment]
breakpoints does not work at all.
the only way to break is to type "debugger" where one need to set the breakpoint.
this way the debugger is invoked and execution stops

**Comment by dancomanlive** [comment]
![screen shot 2018-02-26 at 12 14 28 pm](https://user-images.githubusercontent.com/16872821/36653247-c16b93e2-1aee-11e8-9415-0905aaaff7b1.png)

Getting this error when reloading the application. I don't know if it is relevant to this issue though but I am sharing it just in case it would be. Break point still not working after reinstalling React Native Tools.

**Comment by sergey-akhalkov** [comment]
Hi @Gremlin80, 
>breakpoints does not work at all

Hmm, please let me clarify which way I was able to reproduce the issue:
1. Set up breakpoints
2. Launch `Debug iOS`
3. Got "Breakpoint set but not yet bound" messages
4. Reset one of the breakpoints (just click twice on it)
5. Debugging works again

Here is the demo: ![feb-26-2018 11-04-50](https://user-images.githubusercontent.com/4928157/36659526-d56e1f44-1ae5-11e8-9d9b-0795429fe34a.gif)

So guys, could you please let me know if all of you are experiencing the same issue and if debugging works as expected after resetting breakpoints?

As of now I'm working on fix that should allow to avoid of breakpoints resetting (avoid/remove step 4)

**Comment by netsmertia** [comment]
Yes resetting a single breakpoint allow me to use all the defined breakpoint. 
```
OS: darwin x64
Adapter node: v7.9.0 x64
vscode-chrome-debug-core: 3.23.0
---
netss-MBP:~ nets$ node -v
v9.5.0
---
RN Tools : 0.6.3
react-native-cli: 2.0.1
RN: 0.53.3
VS Code: 1.20.1
```

**Comment by Sler69** [comment]
Yep resetting a single breakpoint enabled all the breakpoints and allowed me to create new ones. Thanks for the feedback @sergey-akhalkov !!

**Comment by novellizator** [comment]
https://mega.nz/#!bQdCXSRQ!aKz9alToOoL4v1z4cJ9Gj3j0_-KGxuhtO1dt15CE9qs
these are my .vscode and .vscode-react-native folders
are you @sergey-akhalkov able to replicate the state of a fucked up debugger with those files please?

**Comment by novellizator** [comment]
I did replicate the bug in two different macs, dunno how I managed that. I had eslint installed on both machines....

**Comment by novellizator** [comment]
I tried the fix of deleting the react native tools (including the .vscode folders in home) and didn't help

**Comment by dancomanlive** [comment]
I got the debugger working but it s only working when causing an action. What I mean is that when I set a breakpoint and reload the app nothing happens but than when I disable the break point and re enable it and than press a button or something the debuggers kicks in. Initially the debugger was working straight away when reloading the app. Hopefully this provides some insight.

**Comment by zmechanic** [comment]
I've got the same issue on VS Code as below:
Version 1.20.1
Commit f88bbf9137d24d36d968ea6b2911786bfe103002
Date 2018-02-13T15:34:36.336Z
Shell 1.7.9
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64

Debugger attaches OK. But all breakpoints are dark grey empty circles:
![image](https://user-images.githubusercontent.com/13178681/36818149-b6b0d270-1cdb-11e8-92b1-b4eea52e669e.png)

**Comment by WHATSAF** [comment]
It seems like not the `vscode-chrome-debug-core`'s issue to me.
I tried the workaround, and it didn't help.
I also switched to lots of versions of the `vscode` and `react-native`, it didn't help too.
```

OS:macOS 10.13.3
VSCode:1.20.1(1.19.x tried)
vscode-chrome-debug-core: 3.23.0
react-native:0.54.0(0.5x.0 tried)
```
Just run `react-native init Rn` and debug.

**Comment by scolestock** [comment]
Can confirm repro of this issue with versions below.  Turning breakpoints off/on again allows for them to trigger.

mac vs code Version 1.20.1 (1.20.1)
react native tools extension 0.6.4
react-native-cli: 2.0.1
react-native: 0.52.0

**Comment by ilkerican** [comment]
Do not use VSCode 1.20 or 1.21. They do not stop at breakpoints.
Use VS Code 1.18 : https://code.visualstudio.com/updates/v1_18 
And dont let your version upgrade by default. Set "update.channel": "none".

Otherwise it updates itself to 1.20 and you go back to unhit breakpoints...

**Comment by yakimka** [comment]
Same here. Breakpoints don't work. Also @sergey-akhalkov workaround also don't work
react-native-tools: 0.6.4
vscode: 1.21.0
os: Arch Linux with latest updates
react-native-cli: 2.0.1
node: 9.8.0
react-native: 0.54.1

these are my .vscode and .vscode-react-native folders https://yadi.sk/d/GO0F50Do3TFHev

**Comment by sergey-akhalkov** [comment]
Hey guys, just want to provide a quick update regarding this issue and also clarify several things:
1. The issue we are talking about in this thread is described here https://github.com/Microsoft/vscode-react-native/issues/625#issuecomment-368421380 and if you have another repro - please create new issue. Currently we are collaborating with the `vscode-chrome-debug-core` team and seems like the root cause of the issue is in `ms-vscode.node-debug2` - we are working on this.
2. @yakimka, the issue you are talking about should be related to this one https://github.com/Microsoft/vscode-react-native/issues/660, @max-mironov already has provided an update regarding this - the issue is in Metro bundler and I think fix should be shipped soon ( 0.54.2 or so )
3. If you are unable to hit breakpoint in other cases (not related to 1st-2nd points) - please make sure you are familar with the following items https://github.com/Microsoft/vscode-react-native#known-issues and https://github.com/Microsoft/vscode-react-native/blob/master/doc/debugging.md otherwise please create separate issue and provide us as much info as possible.

**Comment by pennyandsean** [comment]
0.54.2 is up

**Comment by jdeanwaite** [comment]
Just wanted to chime in and include my report that 0.54.2 does not seem to solve the issue for me (I believe they are still using Metro 0.28.0 instead of 0.29.0).
Receiving the message: `Unverified breakpoint, breakpoints set but not yet bound`
Clicking to reset the breakpoints does nothing.
```
macOS: 10.13.3
React Native Tools 0.6.4
vscode-chrome-debug-core: 3.23.0
```

The interesting thing is that **breakpoints do indeed work with Webstorm** on the same project. Are we positive this is completely an issue with Metro?

**Comment by max-mironov** [comment]
@jdeanwaite we have added tmp fix for this case, could you please upgrade VSCode RN extension for 0.6.5, reload VSCode and try again?

**Comment by jdeanwaite** [comment]
@max-mironov It is working now other than, as you stated, the need to reset a breakpoint once to get it to work (I can live with that).

**Comment by tjma2001** [comment]
This seems to happen to me when I am accessing the project if its in a subdirectory. I have a mono repo with multiple projects. My typescript nodejs application is in a subdirectory called server.

When I open vscode in the subdirectory of server, the breakpoints are showing (i.e. not as unverified breakpoints). They are still not hitting though. When I click run the application, I have a `setTimeout` which prints a `console.log` statement. Before this `setTimeout`, I have another `console.log statement`. In the debugging window when a console.log gets printed it gives you a clickable link to open the file that the `console.log` statement was run from.

![image](https://user-images.githubusercontent.com/8049309/37665219-76f15fee-2c65-11e8-8a32-21fea5c45594.png)

Its only in this file that I can create breakpoints that are actually hit. 

```
Version 1.21.1
Commit 79b44aa704ce542d8ca4a3cc44cfca566e7720f1
Date 2018-03-14T14:46:30.761Z
Shell 1.7.9
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64
```

**Comment by tjma2001** [comment]
I have just changed my debug command from `inspect` to `inspect-brk`. Breakpoints are being hit, but at the wrong points.

This is where is set them:
![image](https://user-images.githubusercontent.com/8049309/37666870-427995a2-2c69-11e8-9956-5b07abbb1f7a.png)

This is where the code is triggering, (once again in the "index.ts `read-only inlined content from source map`")

![image](https://user-images.githubusercontent.com/8049309/37666917-621dadee-2c69-11e8-939b-ecf8934d9595.png)

![image](https://user-images.githubusercontent.com/8049309/37666933-65ec4462-2c69-11e8-8d24-95224a94bc95.png)

This is my current launch.json
```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "attach",
            "name": "Attach to Remote",
            "address": "localhost",
            "port": 5858,
            "protocol": "inspector",
            "localRoot": "${workspaceRoot}",
            "remoteRoot": "/app",
            "sourceMaps": true,
            "restart": true,
            "outFiles": [
                "${workspaceRoot}/server/built/**/*.js"
            ]
        },
        {
            "type": "node",
            "request": "launch",
            "name": "Launch Program",
            "program": "${workspaceFolder}/index.js"
        }
    ]
}
```

**Comment by Benzer1406** [comment]
Still have the same issues with version Version 1.21.1 (1.21.1) of VSCode. With workaround it works, but a fix would be cool.

**Comment by diego-G** [comment]
Same goes here. The weirdest thing is that works on some repos from the same workspace but in the recently settled.

**Comment by snoopy83101** [comment]
@Benzer1406 
Hi ,You say there is a solution to the current version?
What is it?  Pls tell me!  Thank you very much!

**Comment by snoopy83101** [comment]
if  solution is  "Reset a new breakpoint?"
god ! I choose to die

**Comment by Benzer1406** [comment]
Yes @snoopy83101, the workaround is to click on a breakpoint that is already set and deactivate it and then click it again on the line where you want it to be and it works.

**Comment by snoopy83101** [comment]
@Benzer1406 
Whenever i restart a debug
Breakpoint Jumps to an Unknown Line
I must first find the unknown line and cancel him.
Then set it again, in the line I want??


Sorry for the Google translation I used

**Comment by Benzer1406** [comment]
@snoopy83101 then maybe it is a good idea just to use very few breakpoints, Just. One or two.

**Comment by snoopy83101** [comment]
@Benzer1406 
Thank you very much for your answer!
Does the community's big brothers say that this problem will be fixed recently? (in this issues)

**Comment by ak99372** [comment]
None of these "workarounds" do much in my case, all the breakpoints stay with
> Breakpoint set but not yet bound

If I add "debugger;" line into the code the debugger stops at this line however not in code but in "index.bundle"

To me the debugger seems fine but the sourcemaps don't seem to point to the actual code. Is there a way to check and see whether the sourcemap are generated correctly? 

Side node: runing debugger launches 4 Chrome tabs, Is this normal (necessary)?

vscode: 1.21.1
RN tools: 0.6.6
vscode-chrome-debug-core: 3.23.0

**Comment by arvi** [comment]
I am experiencing the same thing "Breakpoint set but not yet bound" :sweat_smile:

**Comment by scolestock** [comment]
Still an issue in VSCode Version 1.21.1 (1.21.1), React Native Tools 0.6.6, 
react-native-cli: 2.0.1
react-native: 0.54.2

**Comment by hippotz** [comment]
This may be a different problem, but I noticed when I upgrade from react-native from 0.54.2 to 0.54.3 my index.map is no longer generated in ./.vscode/.react/

Staying on 0.54.2 is working for me.

My working setup:
OS: Mac OS High Sierra 10.13.4
vscode: 1.21.1
react-native-cli: 2.0.1
react-native: 0.54.2

**Comment by bmr1** [comment]
My breakpoints are grey and never get hit using metro bundler in terminal and using 'Attach to Packager' debug mode.
react-native-cli: 2.0.1
react-native: 0.54.4
windoze 10

**Comment by zhangbobell** [comment]
I also encounter this issue

**Comment by animaonline** [comment]
Upgrading to node 9.11.1 seems to have resolved the problem for me.

**Comment by ASP-2014** [comment]
Problem still persists with VScode Version 1.21.). FYI i have also check on old version (1.17.2). On both version breakpoints are not working at all. Any fix would be really appreciated.

**Comment by rakeshgits** [comment]
Facing same issue with vscode 1.22.1, when i switch to 1.19, it works perfectly, any help will be appreciated !

**Comment by axelnormand** [comment]
Strangely I'm seeing only on my Windows 10 box.
VS Code on my Mac is fine, both setting breakpoints for jest tests and react native debugger seeing the sourcemaps and setting breakpoints.

Both VS Codev1.22.1

react native, typescript, jest, ts-jest, expo

**Comment by mbonaci** [comment]
`1.22.2` behaves the same way.
I'm trying to debug a local `WebpackDevServer`.

**Comment by SudoPlz** [comment]
By the way, Jest debug sessions don't work either now, because the test finishes a lot before I can even manage to click on a breakpoint.

**Comment by Albert-Gao** [comment]
OK. After reading all the previous comments and tried.

2 ways to solve this problem for React native dev:

**1. With the latest version 1.22.2**
1. After starting the app, the breaking point in the app will be black.
2. click it to disable then click again to active it.
3. Now it works.
4. Anytime you refresh, do it again.
5. Still, won't stop inside `render()` function

**2. Downgrade to 1.19**
- Everything works flawlessly
- Remember to turn off the auto update by setting `update.channel` to `none` (Not confirmed)

Otherwise, VSC will auto-update and breaks everything.

**Comment by dancespiele** [comment]
Ok for my doesn't work in the version 1.22.2:

**launch.json:**
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "node",
            "request": "launch",
            "name": "mocha",
            "args": [
                "--recursive",
                "-r",
                "source-map-support/register",
                "-r",
                "ts-node/register",
                "--bail",
                "--watch",
                "${workspaceFolder}/test/**/*.spec.ts",
                "--watch-extensions",
                "ts"
            ],
            "program": "${workspaceFolder}/node_modules/.bin/mocha",
            "preLaunchTask": "tsc: build - tsconfig.json",
            "cwd": "${workspaceFolder}",
            "outFiles": [
                "${workspaceFolder}/lib/**/*"
            ],
            "sourceMaps": true
        }
    ]
}
```

**tsconfig.json**
***json
{
	"compilerOptions": {
		"target": "es5",
		"lib": [
			"es6",
			"dom"
		],
		"module": "commonjs",
		"strict": true,
		"emitDecoratorMetadata": true,
		"experimentalDecorators": true,
		"declaration": true,
		"jsx": "react",
		"jsxFactory": "h",
		"outDir": "./lib",
		"rootDir": ".",
		"sourceMap": true
	},
	"include": [
		"./src",
		"./test"
	],
	"exclude": [
		"node_modules",
		"lib",
		"test"
	]
}
***

I use typescript. When I try to debug, I get the error `Unverified breakpoint, Breakpoints set but not yet bound` in the breakpoints.

**Comment by krazykira** [comment]
@Albert-Gao ``use the debugger statement rather than setting up the VSC`` What do you mean? How to do this ?

**Comment by joebernard** [comment]
@krazykira Instead of setting breakpoints in VSC, add the statement `debugger;` in your code. Run the project in debug mode and the debugger will break at that line.

**Comment by ruslan-bikkinin** [comment]
Hi everyone. 

**TL;DR**
It will be fixed in VS Code v1.24

**Explanation:**
The cause of issue is a [bug](https://github.com/Microsoft/vscode-node-debug2/issues/177) in `vscode-chrome-debug-core` package upon which built-in extension `vscode-node-debug2` depends that we are using. Bug [have fixed](https://github.com/Microsoft/vscode-chrome-debug-core/commit/4883ac884deeab820b709fe41ae1db5013763c69) since **v1.24.0** of `vscode-node-debug2`, but current latest version of it (which vscode is using) is **v1.23.5**. So in the next minor release of VS Code this issue should be fixed automatically.

**Comment by dhalperi** [comment]
When will the next minor release of VSCode be? Sounds like this bug was fixed almost a month ago now.

**Comment by ruslan-bikkinin** [comment]
Hi @dhalperi, as I know, new minor version of VS Code is being released at the first half of each month.

**Comment by ruslan-bikkinin** [comment]
VS Code 1.24 has been released today and this issue has to be resolved now. I tested debugging using this version of VS Code and it worked like a charm, so I am closing this issue. 
Please let us know whether you still experiencing such problem after VS Code update.

**Comment by joebernard** [comment]
I verified it works after the update. Thanks!

**Comment by dxh9845** [comment]
I am still experiencing issues with debug breakpoints not being hit - on version 1.24

**Comment by ruslan-bikkinin** [comment]
Hi @dxh9845 could you please submit a new issue including all the info from the issue template that could help us resolve your problem quicker? Also, please include answers for the following questions:
* Are you running a pure React Native project (not Expo, not using Haul packager)?
* Does your project use javascript or typescript? If it uses typescript, please, let us know how it has been configured for it.
* If it is possible, please, attach demo project.

**Comment by ivanromaniv** [comment]
Hi! I have the same problem. Using 1.24.0 vscode on ubuntu 18.04 with nodejs(javascript) project. I attach to exposed port from docker.

**Comment by ruslan-bikkinin** [comment]
I have locked conversation around this issue due to it's specific symptoms described in the first post. If you are experiencing the similar problem, please, submit a new issue for that.
